src/Build/ElementLocation/ElementLocation.cs

```diff
@@ -71,43 +68,30 @@ public abstract class ElementLocation : IElementLocation, ITranslatable, IImmuta
         /// </summary>
         public string LocationString
         {
-            get { return GetLocationString(File, Line, Column); }
-        }
-
-        /// <summary>
-        /// Gets the empty element location.
-        /// This is not to be used when something is "missing": that should have a null location.
-        /// It is to be used for the project location when the project has not been given a name.
-        /// In that case, it exists, but can't have a specific location.
-        /// </summary>
-        public static ElementLocation EmptyLocation
-        {
-            get { return s_emptyElementLocation; }
+            get
+            {
+                int line = Line;
+                int column = Column;
+                return (line, column) switch
+                {
+                    (not 0, not 0) => ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("FileLocation", File, line, column),
+                    (not 0, 0) => $"{File} ({line})",
+                    _ => File,
+                };
+            }
         }
 
-        /// <summary>
-        /// Get reasonable hash code.
-        /// </summary>
+        /// <inheritdoc />
         public override int GetHashCode()
         {
-            // Line and column are good enough
-            return Line.GetHashCode() ^ Column.GetHashCode();
+            // We don't include the file path in the hash
+            return (Line * 397) ^ Column;
```

Comment:
nit: This should be `unchecked` for perf and in very extreme cases also for correctness.